### PR TITLE
Document user-facing code in rclcpp

### PR DESCRIPTION
**Pull request by jacquelinekay**
With a focus on functions and members used in the demos, such as executors, publish/subscribe and memory strategy classes.

### CODE REVIEW

File: rclcpp/include/rclcpp/executor.hpp [hunk 1]
```diff
@@ -36,6 +36,15 @@ namespace rclcpp
 namespace executor
 {
 
+/// Coordinates the order and timing of available communication tasks.
```
Comment (dirk-thomas) [inline rclcpp/include/rclcpp/executor.hpp]: Verb without `s`: `Coordinate ...`

**Comment by dirk-thomas** [comment]
Currently the detailed descriptions are not extracted by doxygen at all. I assume because they need to start with `/**` instead of `/*` (http://www.stack.nl/~dimitri/doxygen/manual/docblocks.html#cppblock)?

**Comment by esteve** [comment]
+1

**Comment by gerkey** [comment]
I'm resolving conflicts on this now.

**Comment by gerkey** [comment]
This PR only adds/changes comments, so I'm merging it.

**Pull request merged**